File: Data/Entities/Pokemon.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Data.Entities
{
    public class Pokemon
    {
        public int PokemonID { get; set; }
        public int SpeciesID { get; set; }
        public string PokemonName { get; set; }
        public PokemonSpecies Species { get; set; }
    }
}

File: PokedexUI/ClientApp/src/app/models/pokemon.ts
import { PokemonSpecies } from './pokemon-species';

export class Pokemon {
  pokemonID: number;
  pokemonName: string;
  pokemonSpecies: PokemonSpecies;
}

export class CreatePokemon {
  pokemonName: string;
  speciesID: number;
}

export class UpdatePokemon {
  pokemonID: number;
  pokemonName: string;
  speciesID: number;
}
